`d\managed_disk`: Add support for `encryption_settings`

myc2h6o opened the pull request:
`enabled` is to be deprecated per https://github.com/hashicorp/terraform-provider-azurerm/issues/14574#issuecomment-992690956 so not added in the doc

katbyte commented on internal/services/compute/managed_disk_data_source.go:
```diff
@@ -55,6 +55,54 @@ func dataSourceManagedDisk() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"encryption_settings": {
```
settings is redundant in property names
```suggestion
			"encryption": {
```

myc2h6o replied:
in snapshot and managed disk resource it is already named `encryption_settings`. Maybe let's keep it consistent with them?
https://github.com/hashicorp/terraform-provider-azurerm/blob/2443a8844acc475bff373d12ba4858b06561f126/internal/services/compute/managed_disk_resource.go#L187
https://github.com/hashicorp/terraform-provider-azurerm/blob/2443a8844acc475bff373d12ba4858b06561f126/internal/services/compute/snapshot_resource.go#L88

tombuildsstuff replied:
👍 works for me

stephybun reviewed (changes requested):
We also have a test failure
```
------- Stdout: -------
=== RUN   TestAccDataSourceManagedDisk_encryptionSettings
=== PAUSE TestAccDataSourceManagedDisk_encryptionSettings
=== CONT  TestAccDataSourceManagedDisk_encryptionSettings
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: purging of Secret "secret-bf1zk" (Key Vault "https://acctestkv-bf1zk.vault.azure.net/") : keyvault.BaseClient#PurgeDeletedSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=*******;oid=3aa04c8c-5a75-4e5e-9117-1b7cf6f33e21;numgroups=9;iss=https://sts.windows.net/*******/' does not have secrets purge permission on key vault 'acctestkv-bf1zk;location=westeurope'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"ForbiddenByPolicy"}
--- FAIL: TestAccDataSourceManagedDisk_encryptionSettings (289.24s)
FAIL
```

myc2h6o commented:
Hi @stephybun thanks for pointing out the issue. I've fixed the test case and rerun the test cases, please take a look, thanks!
![image](https://user-images.githubusercontent.com/10579712/165922821-1bae4608-959f-4a99-b569-c9ffd3288a57.png)

stephybun reviewed (commented):
Thanks @myc2h6o, the name should still be shortened to `encryption` since `_settings` is implied for the block. This block should be renamed in the other resources as well.

myc2h6o commented:
Got it @stephybun I've renamed it to `encryption` and later I'll update the name in existing resources in a separate pr. The vendor dependency seems breaking, I'll merge main after it's fixed

myc2h6o commented:
Updated test result
![image](https://user-images.githubusercontent.com/10579712/168416172-9885db3c-3568-48aa-83ce-b2a02b8a6a1e.png)

myc2h6o commented:
Hi @tombuildsstuff thanks for reviewing the pr! I've reverted the name back to `encryption_settings`, could you take another look?
Updated test result:
![image](https://user-images.githubusercontent.com/10579712/176589669-273edd46-64d2-4aee-a7c3-1475ec03d667.png)

katbyte reviewed (approved):
LGTM 🎩
